src/SignIn/SignIn.js

```diff
@@ -0,0 +1,26 @@
+import React, { Component } from 'react';
+import authService from '../firebase/authService';
+
+class SignIn extends Component {
+    constructor() {
+        super();
+    }
+
+    signIn() {
+        authService.signIn();
+    }
+
+    signOut() {
+        authService.signOut();
+    }
+
+    render() {
+        if(this.props.isAuthenticated){
+            return <button onClick={this.signOut.bind(this)}>Sign Out</button>
+        }
+
+        return <button onClick={this.signIn.bind(this)}>Sign In</button>  
```

Comment:
to clarify, this is binding the signIn function to this button right? so signIn is called when the button is clicked?

Reply:
Yep. So onClick is a built in React thing that you can pass a callback to.

In React you also have to bind functions, so I just did it here.